scan/lib/scan/test_command_generator.rb

```diff
@@ -111,12 +111,55 @@ def suffix
     def pipe
       pipe = ["| tee '#{xcodebuild_log_path}'"]
 
+      # disable_xcpretty is now deprecated and directs to use output_style of raw
       if Scan.config[:disable_xcpretty] || Scan.config[:output_style] == 'raw'
         return pipe
       end
 
+      formatter = Scan.config[:xcodebuild_formatter].chomp
+      options = legacy_xcpretty_options
+
+      if !options.empty?
+        UI.important("Detected legacy xcpretty being used so formatting wth xcpretty")
+        UI.important("Option(s) used: #{options.join(', ')}")
+        pipe << pipe_xcpretty
+      elsif formatter == 'xcpretty'
+        pipe << pipe_xcpretty
+      elsif formatter == 'xcbeautify'
+        pipe << pipe_xcbeautify
+      else
+        pipe << "| #{formatter}"
+      end
+
+      return pipe
+    end
+
+    def pipe_xcbeautify
+      formatter = ['| xcbeautify']
+
+      if FastlaneCore::Helper.colors_disabled?
+        formatter << '--disable-colored-output'
+      end
+
+      return formatter.join(' ')
```

Comment:
Do need `return`s in new methods? I thought that it's added to stick to code style in this file, but e.g. `actions` doesn't have it.

Reply:
`return` is optional 🤷‍♂️ Ruby will implicitly return the last line in a method but I personally like the explicit `return` 😛